### PR TITLE
Migrate to spf13/cobra for CLI

### PR DESCRIPTION
**Pull request by justaugustus**
#### What type of PR is this?

/kind cleanup feature design

#### What this PR does / why we need it:

This is somewhat minor change, but enables more contributors to potentially work on zeitgeist, given that all of the Release Engineering in https://github.com/kubernetes/release uses cobra (as well as many other Kubernetes codebases).

**Behavior change:**
The previous `local` subcommand has been consolidated into the `validate`
subcommand. `zeitgeist validate` now defaults to a local validation unless the
`--remote` flag is specified (which does both remote and local validation).

/assign @Pluies @saschagrunert @hasheddan @cpanato 
cc: @kubernetes-sigs/release-engineering 

#### Does this PR introduce a user-facing change?

```release-note
- Migrate to spf13/cobra for CLI
  
  **Behavior change:**
  The previous `local` subcommand has been consolidated into the `validate`
  subcommand. `zeitgeist validate` now defaults to a local validation unless the
  `--remote` flag is specified (which does both remote and local validation).
```

### CODE REVIEW

**Review by hasheddan — APPROVED** [review]
/lgtm
/hold because I imagine @Pluies may want to weigh in :)

**Comment by justaugustus** [comment]
Definitely! Thanks for putting the hold on, Dan!

File: cmd/validate.go [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+	rootCmd.AddCommand(validateCmd)
+}
+
+// RunValidate is the function invoked by 'krel gcbmgr', responsible for
+// submitting release jobs to GCB
+func RunValidate(opts *ValidateOptions) error {
+	if opts.Remote {
+		updates, err := dependencies.RemoteCheck(opts.Config)
+		if err != nil {
+			return err
```
Comment (saschagrunert) [inline cmd/validate.go:80]: nit: wrap errors to add additional information context for the user:
```suggestion
			return errors.Wrap(err, "check remote dependencies")
```

**Review by saschagrunert — COMMENTED** [review]
Just some nits here and there, otherwise LGTM :+1:

File: cmd/root.go [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+	logLevel string
+}
+
+var (
+	rootOpts = &rootOptions{}
+
+	// Variables set by GoReleaser on release
+	version = "dev"     // nolint: deadcode,unused,varcheck
+	commit  = "none"    // nolint: deadcode,unused,varcheck
+	date    = "unknown" // nolint: deadcode,unused,varcheck
```
[the remark this answers is not in the document]
Comment (cpanato) [reply cmd/root.go:38]: +1

**Review by cpanato — APPROVED** [review]
/lgtm

File: cmd/root.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (justaugustus) [reply cmd/root.go:40]: Added a TODO for this.

File: .golangci.yml [hunk 3]
```diff
@@ -19,8 +19,6 @@ linters:
     - dogsled
     - dupl
     - errcheck
-    - gochecknoglobals
-    - gochecknoinits
```
[the remark this answers is not in the document]
Comment (justaugustus) [reply .golangci.yml:23]: Added notes in https://github.com/kubernetes-sigs/zeitgeist/pull/11/files#diff-e23b7926f022c85af1665b36d256231bR59.

**Comment by justaugustus** [comment]
Thanks @saschagrunert. Addressed your feedback!

**Review by saschagrunert — APPROVED** [review]
Thanks! 
/lgtm

**Review by Pluies — APPROVED** [review]
That looks great! cobra seems to be the most commonly used option, so good all around 👍

**Comment by Pluies** [comment]
/hold cancel
🥳

**Comment by Pluies** [comment]
Hmm, the label is removed but tide is still unhappy... Let's tryyyy
/retest

**Comment by Pluies** [comment]
Heyyyy, here we go!

**Pull request merged**